src/rhsmlib/services/entitlement.py

```diff
@@ -74,29 +67,13 @@ def get_status(self, on_date: str = None, force: bool = False) -> dict:
             status_cache.server_status = None
             status_cache.delete_cache()
 
-        sorter.load()
-
-        if self.identity.is_valid():
-            overall_status = sorter.get_system_status()
-            overall_status_id = sorter.get_system_status_id()
-            reasons = sorter.reasons.get_name_message_map()
-            reason_ids = sorter.reasons.get_reason_ids_map()
-            valid = sorter.is_valid()
-            status = {
-                "status": overall_status,
-                "status_id": overall_status_id,
-                "reasons": reasons,
-                "reason_ids": reason_ids,
-                "valid": valid,
-            }
-        else:
-            status = {
-                "status": _("Unknown"),
-                "status_id": "unknown",
-                "reasons": {},
-                "reason_ids": {},
-                "valid": False,
-            }
+        status = {
+            "status": _("Unknown"),
+            "status_id": "unknown",
+            "reasons": {},
+            "reason_ids": {},
+            "valid": False,
+        }
```

Comment:
You cannot simplify code like this. This is the reason, why cockpit tests fails.

How to test it?

1. Stop system service provided by rhsm.service: `sudo systemctl stop rhsm.service`
2. Start `rhsm_service.py` using your code: `sudo PYTHONPATH=./src python3 -m subscription_manager.scripts.rhsm_service --verbose` from root of source code
3. Call D-Bus method using this Python method: ``busctl call com.redhat.RHSM1 /com/redhat/RHSM1/Entitlement com.redhat.RHSM1.Entitlement GetStatus ss "" ""

You get always following result despite the system is registered:

```
s "{\"status\": \"Unknown\", \"status_id\": \"unknown\", \"reasons\": {}, \"reason_ids\": {}, \"valid\": false}"
```

Reply:
How to solve this conflict? we have to be compatible with old "constants". In the pure SCA world, there could be only two states "disabled" and "unknown". Thus, we can fix it using:

```python
identity = inj.require(inj.IDENTITY)
if self.identity.is_present():
    status = DISABLED
else:
    status = UNKNOWN
```